app/src/main/java/org/simple/clinic/summary/PatientSummaryScreenController.kt

```diff
@@ -271,69 +305,104 @@ class PatientSummaryScreenController @Inject constructor(
         }
   }
 
-  private fun handleBackAndDoneClicks(events: Observable<UiEvent>): Observable<UiChange> {
-    val callers = events
-        .ofType<PatientSummaryScreenCreated>()
-        .map { it.caller }
-
+  private fun showScheduleAppointmentSheet(events: Observable<UiEvent>): Observable<UiChange> {
```

Comment:
The older method was handling all possible chains from `Back` and `Done` and was getting quite unruly, so I broke it up as a part of this PR. As a consequence, this diff is difficult to parse. Might be easier to open this file separately.

Reply:
Same goes for the `PatientSummaryScreenControllerTest` as well.